### PR TITLE
refactor: update major

### PR DESCRIPTION
**Pull request by dargmuesli**
- update dependencies with major version updates available, migrating `glob`
- migrate vuepress to vitepress

### CODE REVIEW

**Comment by dargmuesli** [comment]
@ota-meshi I think this PR is pretty much ready. But I do not understand why the documentation build fails on Netlify. Maybe you can help me: https://app.netlify.com/sites/heuristic-haibt-5d042e/deploys/6477c19ee446090008dc8d65#L155 :pray:

**Comment by ota-meshi** [comment]
Thank you for great work!
I'll try to find out what the problem is when I have time 👀

**Comment by ota-meshi** [comment]
Hmm. My local build seems to succeed. The version of Node is probably not the cause of the failure either. Maybe it's because of cache.
@kazupon Could you please clear cache and re-run?

**Comment by dargmuesli** [comment]
.. I think I fixed the build? :sweat_smile: Reviews welcome! :heart:

File: docs/rules.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Available Rules
```
Comment (ota-meshi) [inline docs/rules.md:1]: I think the url for this file has changed.
I think we should probably change it to `docs/rules/index.md`.

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: docs/rules.md (hunk 1 above)
Comment (dargmuesli) [reply docs/rules.md]: Changed! :raised_hands:

File: scripts/update-docs-index.ts [hunk 2]
```diff
@@ -39,15 +39,12 @@ ${rules.map(toTableRow).join('\n')}
 `
 }
 
-const filePath = resolve(__dirname, '../docs/rules/README.md')
+const filePath = resolve(__dirname, '../docs/rules.md')
```
Comment (ota-meshi) [inline scripts/update-docs-index.ts:42]: I think we should probably change it.

```suggestion
const filePath = resolve(__dirname, '../docs/rules/index.md')
```

File: docs/rules/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+- :star: mark: the rule which is enabled by `plugin:@intlify/vue-i18n/recommended` preset.
+- :black_nib: mark: the rule which is fixable by `eslint --fix` command.
+
+## Recommended
+
+<!--prettier-ignore-->
+| Rule ID | Description |    |
+|:--------|:------------|:---|
+| [@intlify/vue-i18n/<wbr>no-deprecated-i18n-component](./rules/no-deprecated-i18n-component.html) | disallow using deprecated `<i18n>` components (in Vue I18n 9.0.0+) | :black_nib: |
```
Comment (ota-meshi) [inline docs/rules/index.md:11]: I think we probably need to change the path as well.

```suggestion
| [@intlify/vue-i18n/<wbr>no-deprecated-i18n-component](./no-deprecated-i18n-component.html) | disallow using deprecated `<i18n>` components (in Vue I18n 9.0.0+) | :black_nib: |
```

Comment (dargmuesli) [reply docs/rules/index.md]: done :)

**Review by ota-meshi — APPROVED** [review]
Thank you!

**Pull request merged**